File: SllStack.cpp
#include<iostream>

using namespace std;
class Node{
        int element;
        Node next;
        public:
            Node(int element,Node next){
            element=element;
            next=next;
            }
            int ele(){
                return element;
            }
            Node nod(){
                return next;
            }
};

class SinglyLinkedList{
    
    Node head;
    int size;
    int maxsize;
    public:
        SinglyLinkedList(int max){
            //head=NULL;
            size=0;
            maxsize=max;
        }
    int sizeofList(){
        return size;
    }
    void push(int e){
        if (size<maxsize){
            head=Node(e,head);
            size+=1
        }
    }
    void traverselist(){
        Node n=head;
        int s=
        while (){
            cout<<n.ele()<<endl;
            n=n.nod();
        }
    }
};

int main(){
    SinglyLinkedList sl(5);
    sl.push(3);
    sl.push(4);
    sl.push(5);
    sl.traverselist();
    return 0;
}

File: DSA_Practice/C_programs/SllStack.py
class SinglyLinkedList():
    class _Node:
        __slots__='_element','_next'
        def __init__(self,element, next):
            self._element=element
            self._next=next
    def __init__(self,max):
        self._head=None
        
        self._size=0
        self._maxsize=max
    def __len__(self):
        return self._size
    def push(self,e):
        if self._size<self._maxsize:
            self._head=self._Node(e,self._head)
            self._size+=1
        else:
            raise ValueError("You have reached the max size")
    def top(self):
        return self._head._element
    def traverselist(self):
        n=self._head
        while n is not None:
            print (n._element)
            n=n._next
        print ("\n")
    def secondlast(self):
        n=self._head._next
        while n is not None:
            m=n
            n=n._next._next
        print (m._element)
    def is_empty(self):
        return True if self._head==None else False
    def pop(self):
        if not self.is_empty():
            n=self._head._element
            self._head=self._head._next
            self._size-=1
            return n
        else:
            raise ValueError("List is empty")
    

        
    
sl=SinglyLinkedList(5)
sl.push(5)
sl.push(4)
sl.push(3)
sl.traverselist()
sl.push(2)
sl.push(1)
sl.traverselist()
sl.push(0)
sl.traverselist()

#sl.secondlast()
"""print(sl.pop())
print ("\n")
sl.traverselist()
print(sl.pop())
print ("\n")
sl.traverselist()
print(sl.pop())
sl.traverselist()
print(sl.pop())"""
